src/runtime/utils.ts

```diff
@@ -201,11 +201,11 @@ export async function loadAndSetLocale<Context extends NuxtApp = NuxtApp>(
 
   // set the locale
   if (useCookie) {
-    setCookieLocale(i18n, newLocale)
+    await setCookieLocale(i18n, newLocale)
```

Comment:
I’m not sure if there is any js engine overhead in running as `await` on a synchronous function, but since `setCookieLocale` and `setLocale` are synchronous functions, I think that we don't need to `await`.

Reply:
I think there's no harm in it but rather safer - but I've removed it!